src/nfts/NftImage.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+import { NftEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import SkeletonPlaceholder from 'src/components/SkeletonPlaceholder'
+import { Nft, NftOrigin } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import variables from 'src/styles/variables'
+import Logger from 'src/utils/Logger'
+
+const DEFAULT_IMAGE_HEIGHT = 360
+
+interface ImagePlaceHolderProps {
+  testID: string
+  height?: number
+  width?: number
+  borderRadius?: number
+}
+
+interface Props extends ImagePlaceHolderProps {
+  nft: Nft
+  origin: NftOrigin
+  shouldAutoScaleHeight?: boolean
+  ErrorComponent: React.ReactNode
+}
+
+function ImagePlaceholder({ height = 40, width, borderRadius = 0, testID }: ImagePlaceHolderProps) {
+  return (
+    <SkeletonPlaceholder
+      borderRadius={borderRadius}
+      backgroundColor={colors.gray2}
+      highlightColor={colors.white}
+    >
+      <View
+        style={{
+          height,
+          width: width ?? variables.width,
+          zIndex: -1,
+        }}
+        testID={testID ?? 'NftsInfoCarousel/ImagePlaceholder'}
+      />
+    </SkeletonPlaceholder>
+  )
+}
+
+export default function NftImage({
+  nft,
+  height,
+  width,
+  shouldAutoScaleHeight,
+  borderRadius,
+  origin,
+  ErrorComponent,
+  testID,
+}: Props) {
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState(!nft.metadata)
```

Comment:
Looks like we should combine these 2 in a single `status` state to make it impossible to have it both in loading and error state.

Reply:
makes sense, i updated